target/ppc/helper.h

```diff
@@ -631,6 +631,24 @@ DEF_HELPER_FLAGS_1(tlbia, TCG_CALL_NO_RWG, void, env)
 DEF_HELPER_FLAGS_2(tlbie, TCG_CALL_NO_RWG, void, env, tl)
 DEF_HELPER_FLAGS_2(tlbiva, TCG_CALL_NO_RWG, void, env, tl)
 #if defined(TARGET_PPC64)
+
+/*
+ * tlbie[l] helper flags
+ *
+ * RIC, PRS, R and local are passed as flags in the last argument.
+ */
+#define TLBIE_F_RIC_SHIFT       0
+#define TLBIE_F_PRS_SHIFT       2
+#define TLBIE_F_R_SHIFT         3
+#define TLBIE_F_LOCAL_SHIFT     4
+
+#define TLBIE_F_RIC_MASK        (3 << TLBIE_F_RIC_SHIFT)
+#define TLBIE_F_PRS             (1 << TLBIE_F_PRS_SHIFT)
+#define TLBIE_F_R               (1 << TLBIE_F_R_SHIFT)
+#define TLBIE_F_LOCAL           (1 << TLBIE_F_LOCAL_SHIFT)
+
+DEF_HELPER_FLAGS_4(tlbie_isa300, TCG_CALL_NO_RWG, void, \
```

Comment:
I've noticed now that `helper_tlbie_isa300` may write `env->tlb_need_flush`. Now I'm not sure if it's correct to use `TCG_CALL_NO_RWG` here...

Reply:
It shouldn't be a problem as `env->tlb_need_flush` doesn't seem to be mapped to a TCG global. However, the helper reads `env->msr`, that is mapped as `cpu_msr` in `ppc_translate_init`, so I guess it should be `TCG_CALL_NO_WG`.

Reply:
Hmm, ok, I forgot about the msr access.